src/coreclr/src/jit/importer.cpp

```diff
@@ -4015,6 +4015,36 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Type_get_IsValueType:
+            {
+                // Optimize
+                //
+                //   call Type.GetTypeFromHandle (which is replaced with CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE)
+                //   call Type.IsValueType
+                //
+                // to `true` or `false`
+                // e.g. `typeof(int).IsValueType` => `true`
+                if (impStackTop().val->IsCall())
+                {
+                    GenTreeCall* call = impStackTop().val->AsCall();
+                    if (call->gtCallMethHnd == eeFindHelper(CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE))
+                    {
+                        CORINFO_CLASS_HANDLE hClass = gtGetHelperArgClassHandle(call->gtCallArgs->GetNode());
+                        if (hClass != NO_CLASS_HANDLE)
+                        {
+                            retNode = gtNewIconNode(
+                                (eeIsValueClass(hClass) &&
+                                 // pointers are not value types (e.g. typeof(int*).IsValueType is false)
+                                 info.compCompHnd->getTypeForPrimitiveValueClass(hClass) != CORINFO_TYPE_PTR)
```

Comment:
`asCorInfoType` may be more appropriate than `getTypeForPrimitiveValueClass` here.